src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -109,15 +118,42 @@ public override bool Execute()
             var logger = new MSBuildLoggerAdapter(Log);
             var config = TaskUtilities.TryGetConfig(AnalysisConfigDir, logger);
 
-            if (ShouldMergeAnalysisSettings(Language, config, logger))
+            var languageSettings = GetLanguageSpecificSettings(config);
```

Comment:
The existing code was overly-complicated so I refactored it. There are quite a lot of tests that use this class either directly or indirectly, so I'm confident that the refactoring hasn't impacted any of the existing behaviours.

All of the code paths need to find the language settings, so this is now done up-front.